src/app/modules/shared_models/section_model.nim

```diff
@@ -330,13 +461,17 @@ QtObject:
 
   proc setActiveSection*(self: SectionModel, id: string) =
     for i in 0 ..< self.items.len:
-      if(self.items[i].active):
+      if self.items[i].active:
+        if self.items[i].active == false:
```

Comment:
Hmm, I don't get this part; you already checked that the section is `active` one line above

Reply:
I was tired at the end of the day I guess. I was looking through all the `dataChanged` calls and see if they had `return`s if the value was already the same. I forgot to check the surrounding code correctly on those one. :facepalm: 
Reverting.